packages/unpublished-adapter-encapsulation-test-app/tests/integration/reload-test.js

```diff
@@ -0,0 +1,173 @@
+import { setupTest } from 'ember-qunit';
+import { module, test } from 'qunit';
+import EmberObject from '@ember/object';
+import Store from 'adapter-encapsulation-test-app/services/store';
+import Model, { attr } from '@ember-data/model';
+import JSONAPISerializer from '@ember-data/serializer/json-api';
+import Transform from '@ember-data/serializer/transform';
+import { resolve } from 'rsvp';
+
+class Person extends Model {
+  @attr
+  firstName;
+
+  @attr
+  lastName;
+}
+
+function setupReloadTest(options) {
+  class TestMinimumAdapter extends EmberObject {
+    shouldReloadAllCalled = 0;
+    shouldReloadRecordCalled = 0;
+    shouldBackgroundReloadAllCalled = 0;
+    shouldBackgroundReloadRecordCalled = 0;
+
+    requestsMade = 0;
+
+    constructor() {
+      super(...arguments);
+
+      if (options.shouldReloadAll !== undefined) {
+        this.shouldReloadAll = function() {
+          this.shouldReloadAllCalled++;
+          return options.shouldReloadAll;
+        };
+      }
+
+      if (options.shouldReloadRecord !== undefined) {
+        this.shouldReloadRecord = function() {
+          this.shouldReloadRecordCalled++;
+          return options.shouldReloadRecord;
+        };
+      }
+      if (options.shouldBackgroundReloadAll !== undefined) {
+        this.shouldBackgroundReloadAll = function() {
+          this.shouldBackgroundReloadAllCalled++;
+          return options.shouldBackgroundReloadAll;
+        };
+      }
+
+      if (options.shouldBackgroundReloadRecord !== undefined) {
+        this.shouldBackgroundReloadRecord = function() {
+          this.shouldBackgroundReloadRecordCalled++;
+          return options.shouldBackgroundReloadRecord;
+        };
+      }
+    }
+
+    findAll() {
+      this.requestsMade++;
+      return resolve(options.resolveFindAllWith || { data: [] });
+    }
+
+    findRecord() {
+      this.requestsMade++;
+      return resolve(options.resolveFindRecordWith || { data: null });
+    }
+  }
+  this.owner.register('adapter:application', TestMinimumAdapter);
+
+  this.store = this.owner.lookup('service:store');
+  this.adapter = this.owner.lookup('adapter:application');
+}
+
+module('integration/reload - Reloading Tests', function(hooks) {
+  setupTest(hooks);
+
+  hooks.beforeEach(function() {
+    // Needed to avoid deprecation warning even though not using any transforms.
+    this.owner.register('transform:date', class DateTransform extends Transform {});
```

Comment:
we don't need to register them to avoid this deprecation. There's several nice patterns but probably the best option for this package is to remove the serializer package and register a very simple custom serializer.

Reply:
That may be best, but is complex enough that it would be better to do in a separate PR. What are the other options?

Reply:
After more thought I want us to keep the serializer package in dependencies, and I feel there is a bug in how we determine to throw the transform deprecation. Will investigate separately, let's leave these for now.